compiler/src/compile.re

```diff
@@ -90,21 +90,28 @@ let log_state = state =>
   };
 
 let apply_inline_flags = (prog: Parsetree.parsed_program) => {
```

Comment:
Currently only compiler flags are allowed per-file. Imo I also think this makes more sense than allowing submodules to be compiled differently.

Reply:
Why? A module is a module is a module.

Reply:
I suppose so; I'm having a bit of trouble understanding how some interactions would work with submodules compiled with different flags though; for example say you had:
```
module Main

let mut val = "a"

@compile("no-gc")
module Sub {
  provide let f = () => {
    val = "b"
  }
}

Sub.f()
```
Since `Sub` has gc disabled, would reassigning `val` cause `"a"` to be leaked? Also would `"b"` now be unmanaged? I suppose you have that same problem with file-level flags though and having a GC'd module include a non-GC'd module :thinking:

In any case, allowing submodules to be compiled different would require substantial changes to the existing codebase and I'm not sure I'm confident enough with the module system to be able to get this done before 0.6.0